test/unit/lodash.spec.js

```diff
@@ -0,0 +1,522 @@
+'use strict';
+
+var chai = require('chai');
+var expect = chai.expect;
+var _ = require('lodash');
+var datasets = require('../../app/datasets');
+var Guy = require('./Guy');
+var chance = require('chance').Chance();
+
+describe('Lodash training', function ()
+{
+    describe('Math functions', function () {
+        describe('add', function () {
+
+            var params, capacity, area;
+
+            beforeEach(function () {
+                params = datasets.add();
+                capacity = params[0];
+                area = params[1];
+            });
+
+            it('should verify params', function () {
+                expect(capacity instanceof Function && area instanceof Function).to.eql(true);
+                expect(params).to.have.length(2);
+            });
+
+            it('should return capacity of cube', function () {
+                expect(capacity(2)).to.eql(8);
+                expect(capacity(3.3)).to.eql(35.937);
+            });
+
+            it('should return area of cube', function () {
+                expect(area(2)).to.eql(24);
+                expect(area(3.4)).to.eql(69.36);
+            });
+
+            it('should return add two function', function () {
+                expect(_.add.apply(_, ([capacity(5), area(3)]))).to.eql(179);
+                expect(_.add.apply(_, ([capacity(0.23), area(2)]))).to.eql(24.012167);
+            });
+
+        });
+
+        describe('ceil', function () {
+
+            var params, area;
+
+            beforeEach(function () {
+                params = datasets.ceil();
+                area = params[0];
+            });
+
+            it('should verify params', function () {
+                expect(area instanceof Function).to.eql(true);
+                expect(params).to.have.length(2);
+            });
+
+            it('should return area of rectangle when area side is smaller than 0 return 0', function () {
+                expect(area(3.2,4.6)).to.eql(15.6);
+                expect(area(-2, 4)).to.eql(0);
+                expect(area(0, 0)).to.eql(0);
+                expect(area(-2, 4)).to.eql(0);
+            });
+
+            it('should return round up area of rectangle', function () {
+                expect(_.ceil.apply(_,([area(3.2, 4.4)]))).to.eql(16);
+                expect(_.ceil.apply(_,([area(2.33, 0.55)]))).to.eql(6);
+                expect(_.ceil.apply(_,([area(1.74, 0.54), 4]))).to.eql(4.5601);
+                expect(_.ceil.apply(_,([area(-2, 0.54), 4]))).to.eql(0);
+            });
+        });
+
+        describe('divide', function () {
```

Comment:
Here is a riddle:
Tests are not passing for divide function for provided repository. Why is that happening? (I already know the solution but I would like you to think why that may happen - provided I don't have lodash installed globally)